infrastructure/storage/src/lmdb.rs

```diff
@@ -304,22 +355,28 @@ mod test {
             },
             err => {
                 println!("{:?}", err);
-                assert!(fs::remove_dir_all("./tests/test_overflow").is_ok());
+                assert!(fs::remove_dir_all(test_dir).is_ok());
                 panic!()
             },
         }
-        assert!(fs::remove_dir_all("./tests/test_overflow").is_ok());
+        // Clean up
+        assert!(store.delete_db_from_scope().is_ok());
+        let _no_val = fs::remove_dir_all(test_dir);
+        if std::fs::metadata(test_dir).is_ok() {
+            println!("Database file handles not released, still open in {:?}!", test_dir);
+            assert!(fs::remove_dir_all(test_dir).is_ok());
+        }
     }
 
     #[test]
     fn read_and_write_10k_values() {
-        fs::create_dir("./tests/test_10k").unwrap();
+        let test_dir = "./tests/test_10k";
+        if std::fs::metadata(test_dir).is_ok() {
+            assert!(fs::remove_dir_all(test_dir).is_ok());
+        }
+        fs::create_dir(test_dir).unwrap();
```

Comment:
I think all the create directory calls can be replaced with ```assert!(fs::create_dir(test_dir).is_ok());```

Reply:
Perfect